examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py

```diff
@@ -165,18 +175,44 @@ def train_batch(
 
         return {"loss": loss, "top1_accuracy": top1, "top5_accuracy": top5}
 
-    def evaluate_batch(self, batch: Any, model: nn.Module) -> Dict[str, Any]:
-        input, target = batch
-        logits, _ = model(input)
-        loss = self.criterion(logits, target)
-        top1, top5 = utils.accuracy(logits, target, topk=(1, 5))
+    # def evaluate_batch(self, batch: Any, model: nn.Module) -> Dict[str, Any]:
+    #    input, target = batch
+    #    logits, _ = model(input)
+    #    loss = self.criterion(logits, target)
+    #    top1, top5 = utils.accuracy(logits, target, topk=(1, 5))
+
+    #    return {"loss": loss, "top1_accuracy": top1, "top5_accuracy": top5}
+    def evaluate_full_dataset(
+        self, model, data_loader: torch.utils.data.DataLoader
+    ) -> Dict[str, Any]:
+        acc_top1 = 0
+        acc_top5 = 0
+        loss_avg = 0
+        num_batches = 0
+        with torch.no_grad():
+            for batch in data_loader:
+                batch = self.context.to_device(batch)
+                input, target = batch
+                num_batches += 1
+                logits, _ = model(input)
+                loss = self.criterion(logits, target)
+                top1, top5 = utils.accuracy(logits, target, topk=(1, 5))
+                acc_top1 += top1
+                acc_top5 += top5
+                loss_avg += loss
+        results = {
+            "loss": loss_avg.item() / num_batches,
+            "top1_accuracy": acc_top1.item() / num_batches,
+            "top5_accuracy": acc_top5.item() / num_batches,
+        }
+        if results["top1_accuracy"] > self.results["top1_accuracy"]:
+            self.results = results
 
-        return {"loss": loss, "top1_accuracy": top1, "top5_accuracy": top5}
+        return self.results
```

Comment:
Wouldn't this make validation performance worse for dtrain since `evaluate_full_dataset` is not distributed?

Reply:
The goal here is to provide a better searcher metric for early stopping so that the signal of architecture accuracy is more stable not to validate faster.  Also, the architecture search phase is performed on CIFAR10 and would not benefit much from dtrain.  In the architecture evaluation phase, we do not use evaluate_full_dataset.